contracts/LoanCore.sol

```diff
@@ -743,12 +798,20 @@ contract LoanCore is
         if (protocolFee > 0) _feesWithdrawable[address(this)] += protocolFee;
         if (affiliateFee > 0) _feesWithdrawable[affiliate] += affiliateFee;
 
-        // State changes and cleanup
-        loans[loanId].state = LoanLibrary.LoanState.Repaid;
-        loans[loanId].interestAmountPaid = _amountFromPayer - data.balance;
-        loans[loanId].balance = 0;
+        // state changes
+        if (_paymentToPrincipal == data.balance) {
+            // If the payment is equal to the balance, the loan is repaid.
+            // Full repayments set the _paymentToPrincipal to the balance.
+            // For partial repayments where the amount to pay is greater than the balance,
+            // the _paymentToPrincipal is set to the balance.
```

Comment:
Not sure I follow this comment. The RepaymentController does this, right? But LoanCore is "dumb" to how the RepaymentController works. We already have the greater-than check above, but this seems to be saying that LoanCore is doing something that it is not actually doing.

Reply:
Yes, makes sense. I removed the last three lines of that comment.